### PR TITLE
Interface update

### PR DESCRIPTION
**Pull request by AsianHam**
updated interface to not require an id

determine id on post instead of receiving it form the request body

### CODE REVIEW

File: src/recipe.data.ts [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import mongoose, { Schema } from "mongoose";
 import { Recipe } from "./recipe";
+import { emptyRecipe } from "./emptyRecipe";
+
+type RecipePreview = Omit<Recipe, "id">;
```
Comment (Jacksondr5) [inline src/recipe.data.ts]: You declare this twice

File: src/recipe.data.ts [hunk 2]
```diff
@@ -37,15 +40,35 @@ export function findAllRecipes(): Promise<Array<Recipe> | null> {
   return recipeModel.find().lean().exec();
 }
 
-export function searchForRecipe(ingred: string): Promise<Array<Recipe> | null> {
+export function searchForRecipeIngredient(
+  ingred: string
+): Promise<Array<Recipe> | null> {
```
Comment (Jacksondr5) [inline src/recipe.data.ts]: This should not return null.  If there are no results, the return value should just be an empty array

File: src/recipe.data.ts [hunk 3]
```diff
@@ -37,15 +40,35 @@ export function findAllRecipes(): Promise<Array<Recipe> | null> {
...
     .find({ "ingredients.ingredient": { $regex: `(.*)${ingred}(.*)` } })
     .lean()
     .exec();
 }
 
-export function createOneRecipe(recipeBody: Recipe): Promise<Recipe | null> {
-  return new recipeModel(recipeBody).save();
+export function searchForRecipeName(
+  name: string
+): Promise<Array<Recipe> | null> {
```
Comment (Jacksondr5) [inline src/recipe.data.ts]: Same here

File: src/recipe.data.ts [hunk 4]
```diff
@@ -37,15 +40,35 @@ export function findAllRecipes(): Promise<Array<Recipe> | null> {
...
+  return recipeModel
+    .find({ name: { $regex: `(.*)${name}(.*)`, $options: "i" } })
+    .lean()
+    .exec();
+}
+
+export async function createOneRecipe(
+  recipeBody: RecipePreview
+): Promise<Recipe | null> {
```
Comment (Jacksondr5) [inline src/recipe.data.ts]: When would this return null?

File: src/recipeService.ts [hunk 5]
```diff
@@ -20,6 +21,8 @@ export type RecipeCreationParams = Pick<
   | "steps"
 >;
 
+type RecipePreview = Omit<Recipe, "id">;
```
Comment (Jacksondr5) [inline src/recipeService.ts]: This is repeated again

**Review by Jacksondr5 — CHANGES_REQUESTED** [review]

File: src/recipe.data.ts (hunk 1 above)
Comment (AsianHam) [reply src/recipe.data.ts]: The file needs to be able to reference it.

Are you saying I should export it from the controller and call it that way?

**Review by Jacksondr5 — APPROVED** [review]

**Pull request merged**